### PR TITLE
RHistory window accessability fixes

### LINKED ISSUE #3256: History entries selected by keyboard are unsynced with selected by mouse
- Add 4 entries to history
- Click third entry with mouse (third entry is selected)
- Press up arrow 2 times (first entry is selected)
- Shift+Click on the fourth entry

**Actual:** third and fourth entry are selected
**Expected:** all entries are selected

### LINKED ISSUE #3271: History entry selection can't be expanded using Shift+Arrow
- If entry is selected, Shift+Up/Down should change selected entries
- If text is selected, Shift+Up/Down should change selection similar to the editor
- If several entries are selected, Shift+Up/Down should be similar to Shift+Mouse (see also https://github.com/Microsoft/RTVS/issues/3256)

### LINKED ISSUE #3274: Unselected entries can't be selected using Shift+Click
- Add 4 entries to history
- Select first, third and forth entries with mouse using Ctrl+Click (first, third and forth entries are selected)
- Ctrl+Click on the third entry again (first and forth entries are selected)
- Shift+Click on the third entry

Actual: Nothing is happening
Expected: Third and Forth entries should become selected

### LINKED ISSUE #3381: Plot History device expand/collapse button keyboard focus is hardly identifiable in HC themes
1. Open R Tools > Plots > Plot History Window
2. Switch to R Plot history pane using Ctrl + Tab
3. Using Shift + Alt navigate between pane contents and toolbar
4. Using down arrow key navigate to Device expand/collapse group controls and Observe the keyboard focus in HC black

**Actual Result:**
Expand/collapse group control when focused using keyboard is not identifiable

**Expected Result:**
Expand/collapse group control when focused using keyboard should be identifiable

### PR DESCRIPTION
**Pull request by AlexanderSher**
- Fix #3271: History entry selection can't be expanded using Shift+Arrow
- Fix #3256: History entries selected by keyboard are unsynced with selected by mouse
- Fix #3274: Unselected entries can't be selected using Shift+Click

Attempt to fix #3271 alone has introduced additional bugs. The root cause of all three issues was incorrect ordering of selected entries

- Fix #3381: Plot History device expand/collapse button keyboard focus is hardly identifiable in HC themes

### CODE REVIEW

File: src/R/Components/Impl/History/Implementation/RHistoryEntries.cs [hunk 1]
```diff
@@ -40,40 +49,46 @@ internal abstract class RHistoryEntries : IRHistoryEntries {
...
-        public void Remove(IRHistoryEntry historyEntry) {
-            var entry = historyEntry as Entry;
-            if (entry == null || entry.Owner != this) {
-                throw new ArgumentException(nameof(historyEntry));
+        public void SelectRangeTo(IRHistoryEntry entry) {
+            var rangeEnd = entry as Entry;
+            if (rangeEnd == null) {
+                throw new ArgumentException(nameof(entry));
+            }
+
```
Comment (karthiknadig) [inline src/R/Components/Impl/History/Implementation/RHistoryEntries.cs]: Is this needed? if so shouldn't it be:
var rangeEnd = (Entry)entry; 
That should throw exception if entry is of type Entry.

File: src/R/Components/Impl/History/Implementation/RHistoryEntries.cs [hunk 2]
```diff
@@ -2,28 +2,37 @@
...
+        public IReadOnlyList<IRHistoryEntry> GetSelectedEntries() {
+            var sortedList = new List<Entry>(_selectedStack);
+            sortedList.Sort(EntryComparer);
+            return sortedList;
+        }
+
         public IRHistoryEntry Find(Func<IRHistoryEntry, bool> predicate) => _entries.First(predicate);
         public IRHistoryEntry First() => _entries.First();
         public IRHistoryEntry Last() => _entries.Last();
+        public IRHistoryEntry LastSelected() => _selectedStack[_selectedStack.Count - 1];
```
Comment (MikhailArkhipov) [inline src/R/Components/Impl/History/Implementation/RHistoryEntries.cs]: Never called on empty?

File: src/R/Components/Impl/History/Implementation/RHistoryEntries.cs (hunk 1 above)
Comment (AlexanderSher) [inline src/R/Components/Impl/History/Implementation/RHistoryEntries.cs]: I think I'd rather change the signature to accept line number instead

File: src/R/Components/Impl/History/Implementation/RHistoryEntries.cs (hunk 2 above)
Comment (AlexanderSher) [inline src/R/Components/Impl/History/Implementation/RHistoryEntries.cs]: No. Caller should check `HasSelectedEntries` before calling this method.

**Comment by MikhailArkhipov** [comment]
👍

**Comment by karthiknadig** [comment]
👍

**Pull request merged**